src/linear_algebra/dual.lean

```diff
@@ -0,0 +1,244 @@
+/-
+Copyright (c) 2019 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Fabian Glöckle
+
+Dual vector spaces, the spaces of linear functionals to the base field, including the dual basis
+isomorphism and evaluation isomorphism in the finite-dimensional case.
+-/
+import linear_algebra.dimension
+import linear_algebra.tensor_product
+noncomputable theory
+
+local attribute [instance, priority 0] classical.prop_decidable
+
+namespace linear_equiv
+universes u v w
+variables {α : Type u} {β : Type v} {γ : Type w}
+variables [discrete_field α] [add_comm_group β] [vector_space α β] [add_comm_group γ] [vector_space α γ]
+open vector_space
+
+/-- Version of linear_equiv.dim_eq without universe constraints. -/
+theorem dim_eq_lift (f : β ≃ₗ[α] γ) :
```

Comment:
I think this should move to `linear_algebra.dimension`. I know there are universe constraints there. So just start a new namespace at the bottom that does not have constraints.

Reply:
Okay, done.